### ISSUE TITLE
Can't deploy version 15.2.12 on yocto OS - rocksdb: NotFound: db/ -  _read_fsid unparsable uuid

### ISSUE THREAD

**Issue by insatomcat** [ceph/ceph-ansible#7102]
**What happened:**

I'm trying to use ceph-ansible (stable-5.0) to deploy a ceph cluster on server using a yocto OS (ceph version 15.2.12, the packaged "honister" version: https://layers.openembedded.org/layerindex/recipe/192188/)
The OS already contains the ceph binary.

The playbooks works fine up the the OSD creation part where it fails with an error I could not find much about:

```
Running command: /usr/bin/ceph-osd --cluster ceph --osd-objectstore bluestore --mkfs -i 2 --monmap /var/lib/ceph/osd/ceph-2/activate.monmap --keyfile - --osd-data /var/lib/ceph/osd/ceph-2/ --osd-uuid 0886ca96-9af5-4381-8f17-7924b1ccf5fd --setuser ceph --setgroup ceph
stderr: 2022-03-01T08:03:56.942+0000 740d98abfd00 -1 bluestore(/var/lib/ceph/osd/ceph-2/) _read_fsid unparsable uuid
stderr: 2022-03-01T08:03:56.984+0000 740d98abfd00 -1 rocksdb: NotFound: db/: No such file or directory
stderr: 2022-03-01T08:03:56.984+0000 740d98abfd00 -1 bluestore(/var/lib/ceph/osd/ceph-2/) _open_db erroring opening db:
stderr: 2022-03-01T08:03:57.456+0000 740d98abfd00 -1 bluestore(/var/lib/ceph/osd/ceph-2/) mkfs failed, (5) Input/output error
stderr: 2022-03-01T08:03:57.456+0000 740d98abfd00 -1 OSD::mkfs: ObjectStore::mkfs failed with error (5) Input/output error
stderr: 2022-03-01T08:03:57.457+0000 740d98abfd00 -1 [0;31m ** ERROR: error creating empty object store in /var/lib/ceph/osd/ceph-2/: (5) Input/output error[0m
    --> Was unable to complete a new OSD, will rollback changes
```

**What you expected to happen:**

With a precedent yocto version (yocto dunfell, ceph 15.2.0), I did not have this problem.
This might be linked to ceph but it can also be a problem with the yocto integration, I'm trying to understand what is happening so that I can find the root cause (maybe a link with rocksdb?)

**How to reproduce it (minimal and precise):**

- create a yocto image with honister including ceph (default version will be 15.2.12)
- deploy a ceph cluster (mon, osd, mgr on all nodes, 3 in my setup) with ceph-ansible (branch stable-5.0)

**Share your group_vars files, inventory and **full** ceph-ansibe log**

**Environment**:
* OS (e.g. from /etc/os-release): Yocto honister
* Kernel (e.g. `uname -a`):  5.15.14-rt27-mainline-rt SMP PREEMPT_RT
* Docker version if applicable (e.g. `docker version`): N/A
* Ansible version (e.g. `ansible-playbook --version`): ansible-playbook 2.9.6
* ceph-ansible version (e.g. `git head or tag or stable branch`): stable-5.0
* Ceph version (e.g. `ceph -v`): 15.5.12 (ceph version 128-NOTFOUND (8f69994803975eda09ba6fbec77701982c33af34) octopus (rc))

[ansible.log](https://github.com/ceph/ceph-ansible/files/8159927/ansible.log)
[ceph_group_vars.tar.gz](https://github.com/ceph/ceph-ansible/files/8159931/ceph_group_vars.tar.gz)
[ceph-ansible-site.yaml.gz](https://github.com/ceph/ceph-ansible/files/8159939/ceph-ansible-site.yaml.gz)

Thanks in advance !

**Comment by insatomcat** [comment]
This is a fully virtual environment, ceph being given "/dev/vdb" to create the osd, and those disk being brand new qcow2 files (created with qemu-img create -f qcow2 vm1-osd.qcow2 30G).
I think we can rule out the hardward problem...
Thanks.

**Comment by guits** [comment]
can you show the output of `ls -l /var/lib/ceph/osd/ceph-2/` ?
+ do you have the full ceph-volume.log ?

**Comment by insatomcat** [comment]
This folder is empty (on all 3 nodes).
I'm not sure I have more log for ceph-volume, this command is run by ansible, I guess all the log I have is what gets in ansible.log.
Unless there is something to do to get more logs ?
Thanks.

**Comment by insatomcat** [comment]
ok my problem was this bug: https://tracker.ceph.com/issues/49815
It occurs when using a rocksdb version >= 6.15 (Yocto Honister is using v6.20, Yocto Dunfell is using v6.6).

The fix has been backported to octopus from v15.2.14 only (https://tracker.ceph.com/issues/49981), hence my problem with ceph v15.2.12 + rocksdb 6.20.

Thanks.
